src/main/java/xyz/aadev/generitech/common/blocks/power/BlockPowerStorage.java

```diff
@@ -52,28 +49,27 @@ public BlockPowerStorage() {
 
     @Override
     public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, EntityPlayer player, EnumHand hand, ItemStack heldItem, EnumFacing side, float hitX, float hitY, float hitZ) {
-        TileEntity tileEntity;
-        tileEntity = world.getTileEntity(pos);
-        GeneriTech.Logger.info(Long.toString(((TileEntityPowerStorage) tileEntity).getStoredPower()));
 
         if (!world.isRemote) {
-
             player.openGui(GeneriTech.getInstance(), Reference.GUI_ID.POWERSTORAGE_GUI, world, pos.getX(), pos.getY(), pos.getZ());
-
-
         }
         return true;
     }
 
     @Override
     public IBlockState getActualState(IBlockState state, IBlockAccess worldIn, BlockPos pos) {
-        TileEntityPowerStorage tileEntity = TileHelper.getTileEntity(worldIn, pos, TileEntityPowerStorage.class);
-        if (tileEntity != null && tileEntity.canBeRotated()) {
-            return state.withProperty(FACING, tileEntity.getForward());
+        TileEntity tileEntity = TileHelper.getTileEntity(worldIn, pos, TileEntity.class);
+        if (tileEntity instanceof TileEntityPowerStorage) {
+            if (tileEntity != null && ((TileEntityPowerStorage) tileEntity).canBeRotated()) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Change this condition so that it does not always evaluate to "true" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS2583)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS1066)